strings-csharp/SubstringInString/SubstringInString/SubstringSearchMethods.cs

```diff
@@ -36,24 +33,24 @@ public static List<int> FindAllIndexesWithSpan(string input, string search)
 
         return indexes;
     }
-    public static List<int> FindAllIndexesWithIndexOf(string input, string search)
+
+    public static List<int> FindAllIndexesWithIndexOf(ReadOnlySpan<char> input, ReadOnlySpan<char> search)
     {
         var indexes = new List<int>();
-        var startIndex = 0;
+        var searchLength = search.Length;
+        var inputLength = input.Length;
 
-        do
+        for (int i = 0; i <= inputLength - searchLength; i++)
         {
-            startIndex = input.IndexOf(search, startIndex);
-            if (startIndex != -1)
+            if (input.Slice(i, searchLength).SequenceEqual(search))
             {
-                indexes.Add(startIndex);
-                startIndex++;
+                indexes.Add(i);
             }
         }
-        while (startIndex != -1);
 
         return indexes;
     }
```

Comment:
@lynda-nneka this method has been changed to be identical to the FindAllIndexesWithSpan, it should be changed back to use the IndexOf method like it was using previously.

Reply:
I got a feedback to change it, it's quite confusing taking all these feedbacks as I do not know which exact correction to follow
![Screenshot (45)](https://github.com/CodeMazeBlog/CodeMazeGuides/assets/86114880/d3377620-2e36-4551-a908-b736c24bf3e4)

Reply:
@jshergal why do we need to use plain string here? If we're using the ReadOnlySpan, it should increase performance. Since we won't be copying string when calling the method. Or I'm missing something?

Reply:
@OTybolc and @lynda-nneka,
The issue was not the change to ReadOnlySpan, the code in the method itself was changed and now the two methods are identical and in the IndexOf method, we are not using IndexOf.

@lynda-nneka, the feedback to change to ReadOnlySpan was good feedback, and a suggestion I had made as well, what was confusing is why you also changed the internals of the method to no longer use IndexOf.

![image](https://github.com/CodeMazeBlog/CodeMazeGuides/assets/1489638/bf2a3ca8-c58b-45c0-9c39-6c79a1932ead)

Reply:
@jshergal, 
I didn't have a chance to review the changes yet - just saw that the signature changed. That's why I was a bit confused. Now I see the issue. Thank you.

Reply:
@jshergal  The IndexOf method requires a string comparison type when comparing strings and this overload isnt available for ReadOnlySpan<char>. Except I convert it to a string. What do you think?

Reply:
Hi @lynda-nneka, that is incorrect. There `IndexOf` is available for ReadOnlySpans as well as Spans.
https://learn.microsoft.com/en-us/dotnet/api/system.memoryextensions.indexof?view=net-8.0

Reply:
ok thanks